### ISSUE TITLE
Oauth2 login set to "redirect" doesn't work on all pages

### ISSUE THREAD

**Issue by hpvb** [wekan/wekan#4746]
Not sure if this is a bug or a feature request, but with the oauth2 login style set to "redirect" a direct link to a board still shows a login button instead of directly redirecting.

Since the redirect doesn't happen in this scenario for a user that's not logged in deeplinking to a board/card is effectively broken.

**Comment by xet7** [comment]
@hpvb

Maybe try environment setting OIDC_REDIRECTION_ENABLED=true

Settings added here https://github.com/wekan/wekan/commit/284f4401369aadcec72e67fa935dfc3a9fead721

Originally added here https://github.com/wekan/wekan/pull/4588

**Comment by hpvb** [comment]
It has been set: `      - OIDC_REDIRECTION_ENABLED=true` (in my `docker-compose.yml`) and it works on the main login page. But a deep link to a board like `https://wekan.prehensile-tales.com/b/STRING/BOARDNAME` does not, it displays a login button:
![image](https://user-images.githubusercontent.com/2591099/199624798-d39620ac-a0f2-4ebe-b2ff-26d63dd07e86.png)

**Comment by xet7** [comment]
@hpvb 

Do you mean setting board as Public? Click Private, select Public from dropdown. But then it's visible for everyone at Internet.

Related, if you would like that all boards are private, and can not be set public, select that option at Admin Panel Settings.

**Comment by hpvb** [comment]
No, what I want is that if there's a direct link to a board that authentication happens instead of a user having to click the 'logging in' link. The boards are private on purpose, but everyone inside my organization should be able to see them.

Right now if someone sends a link to a board to someone else in my organization and they aren't yet logged in they get that page, and not a login flow. If they click the 'logging in' link they end up at their dashboard (after the OIDC flow completes) instead of at the clicked link.

So deep linking essentially does not work for users who aren't already logged in. It would be nice if that would work.

Does that make sense like this?

**Comment by xet7** [comment]
@hpvb 

Thanks! So I think, that clicking login should include target url as option like `/sign-in?targeturl` and after login redirect to that page, like mentioned at this router example code. Currently WeKan users FlowRouter. Is that what should happen?

https://stackoverflow.com/questions/32742021/meteor-redirect-after-login-and-logout

**Comment by hpvb** [comment]
@xet7 that would be a huge improvement!

Additionally it would be nice of instead of showing the "logging in" link at all when `OIDC_REDIRECTION_ENABLED=true` it would just do the same thing as if going to the root URL of Wekan. This is less important, clicking the link is inconvenient but not actually broken.

**Comment by xet7** [comment]
@hpvb 

Thanks! So I think, that when user is not logged in, and  `OIDC_REDIRECTION_ENABLED=true`, instead of `Login` link it would redirect to `/sign-in?targeturl`, and after that redirect to that `targeturl` board.

**Comment by hpvb** [comment]
@xet7 yes, and with `OIDC_REDIRECTION_ENABLED=false` it should probably also just redirect? I'm not sure if there's ever a good reason to show that "logging in" link! But that's getting pretty far away from my own use case.

As far as a UI design goes, since the "logging in" link is the only place you *can* go I think the software should just go there. Maybe with an extra message on the login page? Perhaps the logic should be something like this:

* Instead of showing that page you generate a `/sign-in?targeturl` link and redirect to it immediately with a 303 or 401 (Ideally a 401, but I'm not sure how browsers handle that off of the top of my head, maybe test)
* On the `/sign-in` display a message like "The requested board requires authentication" if there is a `targeturl`, if not just show the normal login page
* I think that if this is implemented the current `OIDC_REDIRECTION_ENABLED=true` would take over for my own use case and I think everyone would get a better experience

How does that sound?

**Comment by xet7** [comment]
Moved to #5695
